gammapy/detect/tests/test_test_statistics.py

```diff
@@ -38,3 +39,11 @@ def test_compute_ts_map():
     assert_allclose([[99], [99]], np.where(result.ts == result.ts.max()))
     assert_allclose(6, result.niter[99, 99])
     assert_allclose(1.0227934338735763e-09, result.amplitude[99, 99], rtol=1e-3)
+
+    # test write method
+    with NamedTemporaryFile(suffix='.fits') as f:
```

Comment:
This is not how I've used `NamedTemporaryFile` in other places:
https://github.com/gammapy/gammapy/search?utf8=%E2%9C%93&q=namedtemporaryfile

I don't know how to use it properly, so this is a question: which is better?

The drawback of how you use it is that all code is indented one level more ... if possible I'd like to avoid this.
With the way I use it, it's not clear to me if that is "safe", i.e. when Python opens or closes this temp file, and why I can open it a second time (?) by using the file name.

@adonath – If you know what's going on here, or have time to find out, making this uniform in Gammapy and adding a very short entry to the dev doc HOWTO page would be very helpful.